sqlite/sqlite.go

```diff
@@ -0,0 +1,173 @@
+package sqlite
+
+import (
+	"context"
+	"io/fs"
+	"time"
+
+	"github.com/exoscale/stelling/sqlite/migrationx"
+	"go.uber.org/fx"
+	"zombiezen.com/go/sqlite"
+	"zombiezen.com/go/sqlite/sqlitex"
+)
+
+type Db struct {
+	Path string `validate:"required"`
+}
+
+type DbConfig interface {
+	DbConfig() *Db
+}
+
+func (db *Db) DbConfig() *Db {
+	return db
+}
+
+// NewPool returns a new sqlitex.Pool of connections to the specified sqlite database
+//
+// It will ensure safe defaults are set, such as enabling the WAL and foreign keys
+// Some parameters can be configured by passing in additional `SqliteOption`s
+func NewPool(conf DbConfig, opts ...SqliteOption) (*sqlitex.Pool, error) {
+	sqliteConf := evaluateOptions(conf, opts...)
+
+	poolOpts := sqlitex.PoolOptions{
+		PoolSize: int(sqliteConf.poolSize),
+		Flags:    sqlite.OpenCreate | sqlite.OpenReadWrite | sqlite.OpenWAL,
+	}
+
+	if sqliteConf.inMemory {
+		poolOpts.Flags = poolOpts.Flags | sqlite.OpenMemory
+		poolOpts.PoolSize = 1
+	} else {
+		poolOpts.Flags = poolOpts.Flags | sqlite.OpenURI
+	}
+
+	poolOpts.PrepareConn = func(conn *sqlite.Conn) error {
+		if sqliteConf.busyTimeout != time.Duration(0) {
+			conn.SetBusyTimeout(sqliteConf.busyTimeout)
+		}
+		return sqlitex.ExecuteTransient(conn, "PRAGMA foreign_keys = ON;", &sqlitex.ExecOptions{})
+	}
+
+	return sqlitex.NewPool(sqliteConf.path, poolOpts)
+}
+
+type sqliteConfig struct {
+	path        string
+	poolSize    uint
+	inMemory    bool
+	busyTimeout time.Duration
+}
+
+type SqliteOption func(*sqliteConfig)
```

Comment:
Nit: this stutters (`sqlite.SqliteOption`), so I would rename it to `Option`.

Same for `sqliteConfig`, but that's unexported, so it's less of an issue.

And same for `WithSqliteOptions`, it could be `WithOptions`.

Reply:
The idea was to explicitly distinguish it from the `ModuleOption`.

Renaming the `SqliteOption` to `Option` is probably fine, but renaming `WithSqliteOption` to `WithOption` feels like it's making the api more confusing to me.